platform/linux-generic/pktio/netmap.c

```diff
@@ -786,6 +824,100 @@ static int netmap_recv(pktio_entry_t *pktio_entry, int index,
 	return num_rx;
 }
 
+static int netmap_recv_tmo(pktio_entry_t *pktio_entry, int index,
+			   odp_packet_t pkt_table[], int num, uint64_t usecs)
+{
+	struct timeval timeout;
+	int ret;
+	int maxfd;
+	fd_set readfds;
+
+	ret = netmap_recv(pktio_entry, index, pkt_table, num);
+	if (ret != 0)
+		return ret;
+
+	/* 32 bit architectures use 32-bit division if possible */
+	if (sizeof(size_t) < 8) {
+		if (usecs <= UINT32_MAX) {
+			timeout.tv_sec = ((uint32_t)usecs) / (1000 * 1000);
+			timeout.tv_usec = ((uint32_t)usecs) % (1000 * 1000);
+		} else {
+			timeout.tv_sec = usecs / (1000 * 1000);
+			timeout.tv_usec = usecs % (1000 * 1000);
+		}
+	} else {
+		timeout.tv_sec = usecs / (1000 * 1000);
+		timeout.tv_usec = usecs % (1000 * 1000);
```

Comment:
The second division can be avoided.
```
timeout.tv_usec = usecs - (timeout.tv_sec * 1000000);
```

Reply:
Strange how I didn't realize this... But this definitely will be in the next patch version!